### PR TITLE
Updating stepper to allow new step value

### PR DESCRIPTION
**Pull request by richardstevens**
#### What does this PR do? (please provide any background)
This will allow someone to specify a step value to use when incrementing or decrementing the stepper value.
#### What tests does this PR have?
`npm run test`
#### How can this be tested?
Pull down the stepper-stepvalue branch and you can then run the `npm run test` and `npm run coverage` and `npm run lint` in order to check it passes them all for stepper.

---
#### Developer Definition of Done/Quality Checklist (for PR author to complete BEFORE code review):
- [X] I have checked the [contributing document](../blob/master/CONTRIBUTING.md) and I'm happy for this to be reviewed.

#### Software Engineer or Developer review:
- [x] I’ve witnessed the work behaving as expected (this could be on the authors machine or screencast).
- [x] I’ve checked for coding anti-patterns.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked all the tests are passing.

#### Software Engineer or project guru review:
- [x] I’ve witnessed the work behaving as expected (this could be on the authors machine or screencast).
- [x] I’ve checked for coding anti-patterns.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked all the tests are passing.

### CODE REVIEW

**Comment by elliottcrush** [comment]
Everything working as expected, appropriate test coverage and have seen the compiled result in a working example.

File: src/components/stepper/stepper.jsx [hunk 1]
```diff
@@ -8,30 +8,33 @@ module.exports = React.createClass({
...
-    onChange: React.PropTypes.func,
+    onChange: React.PropTypes.func.isRequired,
     minValue: React.PropTypes.number,
     maxValue: React.PropTypes.number,
+    stepValue: React.PropTypes.number,
     incrementDisplayString: React.PropTypes.string,
-    decrementDisplayString: React.PropTypes.string
+    decrementDisplayString: React.PropTypes.string,
+    id: React.PropTypes.string,
+    label: React.PropTypes.string
```
Comment (jodiedoubleday) [inline src/components/stepper/stepper.jsx]: YES! good job on including a label!

File: src/components/stepper/stepper.jsx [hunk 2]
```diff
@@ -8,30 +8,33 @@ module.exports = React.createClass({
...
     value: React.PropTypes.number,
-    onChange: React.PropTypes.func,
+    onChange: React.PropTypes.func.isRequired,
     minValue: React.PropTypes.number,
     maxValue: React.PropTypes.number,
+    stepValue: React.PropTypes.number,
     incrementDisplayString: React.PropTypes.string,
-    decrementDisplayString: React.PropTypes.string
+    decrementDisplayString: React.PropTypes.string,
+    id: React.PropTypes.string,
```
Comment (jodiedoubleday) [inline src/components/stepper/stepper.jsx]: I hope ID is used just for usability (label - for) rather an used to style.

File: src/components/stepper/stepper.jsx [hunk 3]
```diff
@@ -51,7 +54,7 @@ module.exports = React.createClass({
           <Button handleClick={this.decrement} type="button" disabled={!this.canDecrement()}>{this.props.decrementDisplayString}</Button>
         </span>
 
-        <Input type="text" key={this.props.value} readOnly={true}>{this.props.value.toString()}</Input>
+        <Input type="text" id={this.props.id} key={this.props.value} readOnly={true} label={this.props.label}>{this.props.value.toString()}</Input>
```
Comment (jodiedoubleday) [inline src/components/stepper/stepper.jsx]: Not sure label is used right here. I was expecting a label to be passed into the stepper component and a `<label for="#">` to be put in front of the stepper. Is this right?

File: src/components/stepper/stepper.jsx (hunk 2 above)
Comment (richardstevens) [inline src/components/stepper/stepper.jsx]: Yeah just for the label to reference the input in the for="" attribute

File: src/components/stepper/stepper.jsx (hunk 3 above)
Comment (richardstevens) [inline src/components/stepper/stepper.jsx]: This just passes the label text through to the input component which will then create the label and use the text passed through. I was surprised it was this way around but it solved how i needed it to work so didn't want to mess with the main Input component

Comment (jodiedoubleday) [inline src/components/stepper/stepper.jsx]: Ahhh of course its using the INPUT component which creates the label for you.

Awesome stuff!

**Comment by richardstevens** [comment]
Excellent thanks

**Pull request merged**